### PR TITLE
Upgrade to mypy 0.790

### PR DESCRIPTION
**Pull request by gpotter2**
- upgrade to mypy 0.790

### CODE REVIEW

File: scapy/main.py [hunk 1]
```diff
@@ -258,7 +258,7 @@ def list_contrib(name=None,  # type: Optional[str]
             continue
         if mod.endswith(".py"):
             mod = mod[:-3]
-        desc = {"description": None, "status": None, "name": mod}
+        desc = {"description": "", "status": "", "name": mod}
```
Comment (gpotter2) [inline scapy/main.py:261]: The `sort` call on line 284 requires the value to be comparable. On Python3, `None` isn't comparable.

File: scapy/layers/can.py [hunk 2]
```diff
@@ -380,21 +381,20 @@ def __iter__(self):
 
     @staticmethod
     def open(filename):
-        # type: (Union[IO[Any], str]) -> Tuple[str, IO[Any]]
+        # type: (Union[IO[bytes], str]) -> Tuple[str, _ByteStream]
```
Comment (gpotter2) [inline scapy/layers/can.py:384]: Same than what was done in `scapy/utils.py`

**Comment by gpotter2** [comment]
@guedou Could you also please review this one

File: .github/workflows/unittests.yml [hunk 3]
```diff
@@ -44,7 +44,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v2
         with:
-          python-version: 3.9
```
Comment (gpotter2) [inline .github/workflows/unittests.yml:47]: This was downgraded because 3.9 raised some `DeprecationWarnings` within mypy.

**Review by guedou — APPROVED** [review]

**Pull request merged**